MantidPlot/src/Mantid/MantidDock.cpp

```diff
@@ -404,8 +404,8 @@ void MantidDockWidget::setItemIcon(QTreeWidgetItem *item,
 */
 void MantidDockWidget::updateTree() {
   // do not update until the counter is zero
-  if (m_updateCount.deref())
-    return;
+  // if (m_updateCount.deref())
+  //   return;
```

Comment:
I am not sure that this is the best fix (although I am not familiar enough with `MantidDock` to know what the best fix would be):

As the comment above says, the `if` above avoids updating the tree until the counter is zero. I have been debugging the code with this `if` block enabled (and keeping you other changes) and I think the problem appears when `m_updateCount ==0`. In that case, according to Qt documentation, `m_updateCount.deref()` will:

```
Atomically decrement the value of this QAtomicInt. Return true if the new value is non-zero, false otherwise
```

but as the new value is `-1` it will return `true`. Instead of removing the `if` block, I've tried (keeping your other changes):

```
  // do not update until the counter is zero
	m_updateCount.deref();
	if (int(m_updateCount) > 0) {
		return;
	}
```

and that seems to work too. However I can't be sure that it's the right think to do, therefore I am leaving this PR unassigned so that other developer can test it.